src/Type/CompoundTypeHelper.php

```diff
@@ -0,0 +1,21 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type;
+
+class CompoundTypeHelper
+{
+
+	public static function accepts(CompoundType $compoundType, Type $otherType): bool
+	{
+		if ($compoundType instanceof MixedType) {
+			return true;
+
+		} elseif ($compoundType instanceof UnionType) {
+			return UnionTypeHelper::acceptsAll($otherType, $compoundType);
+
+		} else {
+			throw new \LogicException();
```

Comment:
I prefer throwing `\PHPStan\ShouldNotHappenException` exception on theoretically unreachable lines. You can also remove the `else` branch and un-indent this line.